kong/plugins/datadog/handler.lua

```diff
@@ -128,25 +137,23 @@ local function log(premature, conf, message)
     request_count    = 1,
   }
 
-  local logger, err = statsd_logger:new(conf)
-  if err then
-    ngx_log(NGX_ERR, "failed to create Statsd logger: ", err)
-    return
-  end
-
   for _, metric_config in pairs(conf.metrics) do
     local metric = metrics[metric_config.name]
 
-    if metric then
-      metric(name, message, metric_config, logger)
+    local tags = metric_config.tags or {}
```

Comment:
I did this at your request but I believe that the config can live from request to request and seems to build up. Is that true?

Reply:
Sorry, can you elaborate more?

Reply:
Well, if there are tags on the config then this is just a reference to the tags on the config. It then gets modified when we insert the api name. If the reference isn't cleared between requests then every request will append more to the table.

Something i noticed when working locally but wasn't sure if i was interpreting it correctly.

Reply:
I see it now, but @thibaultcha correct me if I am wrong, a new copy of plugin config is pulled for each request from the cache, so making change to config in one request should not affect other request.

Reply:
A plugin's `config` outlives a single requests - yes, it is shared between multiple ones, as it is cached in the Lua-land (and should be read-only, in the future).

Reply:
@kbacha mind updating the code to use copy of tag table? `kong.tools.utils` has a `shallow_copy` function.

Reply:
For sure!

Reply:
Added a test to demonstrate what happens and used shallow copy to correct it.